### PR TITLE
Notebook scrolling to search result instead of selected cell

### PR DESCRIPTION
**Pull request by barbaravaldez**
We need to check that the **current match** search result is an active cell before scrolling, to make sure that we don't scroll when clicking on another cell.
Chatted with @MaddyDev offline and decided to remove `setCurrentFindMatch` from the listener because it was overwriting the every time we set a new active cell to the current search result, instead.

This PR fixes https://github.com/microsoft/azuredatastudio/issues/20488

https://user-images.githubusercontent.com/34872381/194933861-e813a35f-6311-4598-8188-708a449e75f7.mov

### CODE REVIEW

**Comment by Charles-Gagnon** [comment]
Please add a gif of new behavior

**Comment by lucyzhang929** [comment]
If cell 1 is active and I search for a term that is in cell 2, I would still expect the notebook to scroll to cell 2 even though cell 1 is active. Is that not what we want?

**Comment by MaddyDev** [comment]
+1 to Lucy's comment. Check why addDecoration is being called when you're selecting a cell and correcting that behavior is the right way to go here.

**Comment by barbaravaldez** [comment]
> +1 to Lucy's comment. Check why addDecoration is being called when you're selecting a cell and correcting that behavior is the right way to go here.

@MaddyDev every time we focus on a cell the updatePreview method is called. I checked the scenario that Lucy mentioned above and it didn't work before this change. Since updatePreview is not being called when pressing enter on the findWidget

**Comment by MaddyDev** [comment]
> > +1 to Lucy's comment. Check why addDecoration is being called when you're selecting a cell and correcting that behavior is the right way to go here.
> 
> @MaddyDev every time we focus on a cell the updatePreview method is called. I checked the scenario that Lucy mentioned above and it didn't work before this change. Since updatePreview is not being called when pressing enter on the findWidget

What do you mean it doesn't work? I just tested in insiders and it works. When a cell is active and you search for a term (that's in another cell), it needs to scroll to the cell that has the search term.

**Comment by barbaravaldez** [comment]
Closing this for now. There's a regression on the current scroll to find result logic that needs to be fixed first.

**Comment by Charles-Gagnon** [comment]
Also - I just noticed that the output component also has very similar looking find logic. Do you have to fix that up too?

https://github.com/Microsoft/azuredatastudio/blob/d3c474162d48807901b3034b54794d4c634daf15/src/sql/workbench/contrib/notebook/browser/cellViews/output.component.ts#L204

File: src/sql/workbench/contrib/notebook/browser/cellViews/interfaces.ts [hunk 1]
```diff
@@ -73,18 +73,23 @@ export abstract class CellView extends AngularDisposable implements OnDestroy, I
...
 						each: function (node, range) {
 							// node is the marked DOM element
-							node.scrollIntoView({ block: 'center' });
+							highlightedNode = node;
+
 						}
 					});
+
+					if (range.cell.active) {
+						highlightedNode.scrollIntoView({ block: 'center' });
```
[the remark this answers is not in the document]
Comment (barbaravaldez) [reply src/sql/workbench/contrib/notebook/browser/cellViews/interfaces.ts:91]: I tried this but the scrolling stops working. We handle scrolling for text cells and code cells separately so maybe the other node was a code cell?

[the remark this answers is not in the document]
Comment (Charles-Gagnon) [reply src/sql/workbench/contrib/notebook/browser/cellViews/interfaces.ts:91]: Ah shoot, yeah it was a code cell. 

But my question still stands though - how is it scrolling if we're checking the active cell? Or is the intent now that if the user has a cell selected we won't scroll to the find results even if they click enter/arrow buttons?

[the remark this answers is not in the document]
Comment (MaddyDev) [reply src/sql/workbench/contrib/notebook/browser/cellViews/interfaces.ts:91]: I agree with Charles here, lets move the scroll check to each.

[the remark this answers is not in the document]
Comment (barbaravaldez) [reply src/sql/workbench/contrib/notebook/browser/cellViews/interfaces.ts]: We update the decorations when the find widget is active and the text cell modified. The problem is that updatePreview is called every time we focus on a new cell. To fix this we only scroll if search result cell is active, which is set [here](https://github.com/microsoft/azuredatastudio/blob/main/src/sql/workbench/contrib/notebook/browser/notebookEditor.ts#L471)

**Comment by MaddyDev** [comment]
Could you update the gif with current fixes.

**Review by MaddyDev — APPROVED** [review]

File: src/sql/workbench/contrib/notebook/browser/cellViews/interfaces.ts [hunk 2]
```diff
@@ -73,16 +73,18 @@ export abstract class CellView extends AngularDisposable implements OnDestroy, I
 				let elements = this.getHtmlElements();
 				if (elements?.length >= range.startLineNumber) {
 					let elementContainingText = elements[range.startLineNumber - 1];
+					let isCellActive = range.cell.active;
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/notebook/browser/cellViews/interfaces.ts:76]: Could just move this down into the if

`if (range.cell.active)`

**Review by Charles-Gagnon — APPROVED** [review]

**Pull request merged**